internal/services/oracle/oracle_autonomous_database_clone_from_database_resource.go

```diff
@@ -0,0 +1,492 @@
+				CharacterSet:                   pointer.To(model.CharacterSet),
+				ComputeCount:                   pointer.To(model.ComputeCount),
+				ComputeModel:                   pointer.To(autonomousdatabases.ComputeModel(model.ComputeModel)),
+				CustomerContacts:               pointer.To(expandCloneCustomerContacts(model.CustomerContacts)),
+				DataStorageSizeInTbs:           pointer.To(model.DataStorageSizeInTb),
+				DbWorkload:                     pointer.To(autonomousdatabases.WorkloadType(model.DatabaseWorkload)),
+				DbVersion:                      pointer.To(model.DatabaseVersion),
+				DisplayName:                    pointer.To(model.DisplayName),
+				IsAutoScalingEnabled:           pointer.To(model.AutoScalingEnabled),
+				IsAutoScalingForStorageEnabled: pointer.To(model.AutoScalingForStorageEnabled),
+				IsMtlsConnectionRequired:       pointer.To(model.MtlsConnectionRequired),
+				LicenseModel:                   pointer.To(autonomousdatabases.LicenseModel(model.LicenseModel)),
+				NcharacterSet:                  pointer.To(model.NationalCharacterSet),
+				WhitelistedIPs:                 pointer.To(model.AllowedIps),
+			}
+
+			properties := param.Properties.(*autonomousdatabases.AutonomousDatabaseCloneProperties)
+			if model.RefreshableModel != "" {
+				properties.RefreshableModel = pointer.To(autonomousdatabases.RefreshableModelType(model.RefreshableModel))
+			}
+
+			if model.SubnetId != "" {
+				properties.SubnetId = pointer.To(model.SubnetId)
+			}
+
+			if model.VnetId != "" {
+				properties.VnetId = pointer.To(model.VnetId)
+			}
+
+			if err := client.CreateOrUpdateThenPoll(ctx, id, param); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (r AutonomousDatabaseCloneFromDatabaseResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Oracle.OracleClient.AutonomousDatabases
+			id, err := autonomousdatabases.ParseAutonomousDatabaseID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			var state AutonomousDatabaseCloneResourceModel
+
+			if v, ok := metadata.ResourceData.GetOk("refreshable_model"); ok {
+				state.RefreshableModel = v.(string)
+			}
+
+			if model := resp.Model; model != nil {
+				state.Location = location.Normalize(model.Location)
+				state.Tags = pointer.From(model.Tags)
+				state.Name = id.AutonomousDatabaseName
+				state.ResourceGroupName = id.ResourceGroupName
+
+				props, ok := model.Properties.(autonomousdatabases.AutonomousDatabaseCloneProperties)
+				if !ok {
+					return fmt.Errorf("%s was not of type `Clone`", id)
+				}
```

Comment:
Please change it as below:

if props := model.Properties; props != nil {
...
}